calcom/cal.com
# feat: #1490 [CAL-422] Intercom App

## What does this PR do?

This PR introduces an integration between Intercom and Cal.com, aimed at simplifying the process of scheduling meetings. Leveraging Intercom's OAuth framework, users can now seamlessly configure and utilize a "Book a Meeting" button directly within their Intercom Messenger. This integration enhances communication efficiency and offers users a convenient way to schedule meetings hassle-free. The PR encompasses OAuth authentication, configurable scheduling options, and real-time updates for a seamless user experience.

Fixes #1490 
/claim #1490

## Requirement/Documentation

To understand and implement the purpose of this PR, follow these steps:

1. **Cal.com App Setup:**
   - Create a [Cal.com app](https://developers.intercom.com/building-apps/docs).
   
2. **Intercom Configuration:**
   Activate and configure your app in Intercom's ***Authentification > OAuth***:
   - **Redirect:** Use `https://CAL_URL/api/integrations/intercom/callback`.
 
   Configure your app in Intercom's ***Canvas Kit > For Users, Leads, and Visitors***:
   
   - **Checkboxes:** Select relevant checkboxes (minimum one).
   - **Initialize Webhook:** Use `https://CAL_URL/api/integrations/intercom/initialize`.
   - **Submit Webhook (mandatory but not used):** Use `https://CAL_URL/api/integrations/intercom/submit`.
   - **Configure Webhook:** Use `https://CAL_URL/api/integrations/intercom/configure`.
   - **[EDIT] Submit Sheet Webhook** Use `https://CAL_URL/api/integrations/intercom/sheet`. 
   
4. **Admin Activation:**
   The admin account is responsible for activating the Intercom app in Cal dashboard. This involves providing the Client Id and Client Secret (available in the ***Authentication*** section of your Intercom app's configuration settings).

This PR facilitates OAuth integration and enables the "Book a Meeting" button in Intercom Messenger. Follow these steps meticulously for a seamless Cal.com and Intercom integration.

## Type of change


## How should this be tested?

To verify the changes made by this PR, follow these steps:

1. **Intercom App Configuration:**
   - If not already done, refer to the previous section to set up your Cal.com app in Intercom.

2. **OAuth Authentication:**
   - Log in to the Cal.com app as a user.
   - Navigate to the Cal.com app's store.
   - Activate the Intercom app using OAuth.

3. **Intercom Dashboard:**
   - Customize your [Messenger](https://www.intercom.com/help/en/articles/6612589-set-up-and-customize-the-messenger#h_0d6e0fb3cf) with the Cal.com App (if you checked the Messenger box before).
   - Save and set it live.
   - Integrate your Intercom Messenger into a test website.

5. **User Experience on Tested Website:**
   - Engage in a conversation as a user or lead in Intercom.
   - Notice the presence of the "Book a Meeting" button in the messenger.
   - Click the button and verify if it redirects to the Cal.com booking page.


By following these steps, you can validate the integration of the Cal.com Intercom app and ensure its smooth functionality.


## Mandatory Tasks


## Checklist

- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works
